play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/accordion/Accordion.scala

```diff
@@ -54,5 +54,18 @@ object Accordion {
       (__ \ "rememberExpanded").readNullable[Boolean]
   )(Accordion.apply _)
 
-  implicit def jsonWrites: OWrites[Accordion] = Json.writes[Accordion]
+  implicit def jsonWrites: OWrites[Accordion] = (
+    (__ \ "id").write[String] and
```

Comment:
What's made us require an explicitly defined `OWrites` here, rather than the automatic one? Can't see it but interested to know.

Reply:
For the other question around OWrites, we need a custom writer here because the case class has fields without the text as a suffix, but the JSON examples from the nunjucks has the text as a suffix, so to keep it consistent and get tests to pass we always ensure its written with the text suffix

Reply:
Basically we've used different fieldnames in our [viewmodel](https://github.com/hmrc/play-frontend-hmrc/blob/5cf0fb5656ce527eae517fe422d15b053c5be8cb/play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/accordion/Accordion.scala#L29-L34) than the names used by [govuk-frontend](https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/components/accordion/accordion.yaml#L22-L44), so without a custom writer to fix up the field names, the nunjucks component won't see any of those fields.

Ideally I'd like to deprecate the existing, incorrect, field names and realign them with govuk-frontend, but we said we'd kick that into tech debt for now.